src/internal/hooks/use-visual-mode/index.ts

```diff
@@ -21,6 +26,9 @@ function detectVisualRefresh() {
 
 export function useVisualRefreshDynamic() {
   if (visualRefreshState === undefined) {
+    if (typeof window !== 'undefined' && window[Symbol.for('isVisualRefresh')]?.()) {
+      document.body.classList.add('awsui-visual-refresh');
+    }
     visualRefreshState = detectVisualRefresh();
```

Comment:
Can we change the order of these checks and only read the symbol value if class name is not found?

This is needed for backward compatibility and also to prevent multiple access to the symbol when multiple instances load on the page. Only the first one will read the symbol, all following get it from the DOM, as before